Documentation/scalability/scalability.rst

```diff
@@ -0,0 +1,397 @@
+see an increase of the **Endpoint regeneration time** which **peaked a little
+above 3s.**
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+10. "Accidentally" destroy 10000 pods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this step we have "accidentally" destroyed 10000 random pods. Kubernetes will
+then recreate 10000 new pods so it will help us understand what is the
+convergence time for all the network policies deployed.
+
+.. figure:: images/image_10_01.png
+.. figure:: images/image_10_02.png
+
+
+* The first mark represents the point in time when pods were "deleted" and the
+  second mark represents the point in time 10k new pods finished to be
+  recreated.
+
+* Besides the CPU usage slightly increasing while pods are being scheduled in
+  the cluster we did see some interesting data points in the BPF memory usage.
+  As each endpoint can have one or more dedicated BPF maps, the BPF memory usage
+  is directly correlated with the number of pods running in a node. **If the
+  number of pods per node decreases so does the BPF memory usage.**
+
+.. figure:: images/image_10_03.png
+
+Another interesting data point is the Endpoint regeneration time is "inexistent".
+Unfortunately, we were not able figure out why the metric was not available for
+this time period. Luckily enough we had another metric that was showing how
+many Cilium endpoints had policy being enforced:
+
+.. figure:: images/image_10_04.png
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+11. Control plane metrics over the test run
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The focus of this test was to study the Cilium agent resource consumption at
+scale. However we also monitored some metrics of the control plane nodes such as
+etcd metrics and CPU usage of the k8s-controllers and we present them in the
+next figures.
+
+.. figure:: images/image_11_01.png
+
+Memory consumption of the 3 etcd instances during the entire scalability
+testing.
+
+.. figure:: images/image_11_02.png
+
+CPU Usage for the 3 controller nodes, average latency per request type in
+the etcd cluster as well as the number of operations per second made to etcd.
+
+.. figure:: images/image_11_03.png
+
+All etcd metrics, from left to right, from top to bottom: database size,
+disk sync Duration, client traffic in, client traffic out, peer traffic in,
+peer traffic out.
+
+=============
+Final Remarks
+=============
+
+With this test we were able to have a better understanding of Cilium running
+in a large cluster entirely in CRD mode and without depending on etcd. There is
+still some work to be done with regarding the memory footprint of BPF maps which
+can be optimized even further as well as the memory footprint of the Cilium
+agent which we will address in the next Cilium version.
+
+We can also determine that it is scalable to run Cilium in CRD mode on a cluster
+with more than 200 nodes. However it is worth to point out that more tests will
```

Comment:
maybe:

However, it is worth pointing out that we need to run more tests to verify Cilium's behavior when it loses the connectivity with kube-apiserver, which can happen during a control plane upgrade for example.